### DISCUSSION TITLE
mariaDB to postgreSQL: Violates not-null constraint

### DISCUSSION THREAD

**Discussion by aprakasa** [adonisjs/core#1942] [General]
I am trying to migrate my current database mariaDB to postgreSQL.
But the POST request always get an error after the migration:
```
"insert into \"counters\" (\"created_at\", \"is_active\", \"name\", \"updated_at\") values ($1, $2, $3, $4) returning \"id\" - null value in column \"id\" of relation \"counters\" violates not-null constraint"
```

**Comment by thetutlage** [comment]
Need more information to help further. At bare minimum share the migration file for the users table

**Reply by aprakasa** [reply]
Sorry, first time doing migration. Is more about the postgres behavior.
Find the solution here https://kylewbanks.com/blog/Adding-or-Modifying-a-PostgreSQL-Sequence-Auto-Increment

**Reply by aprakasa** [reply]
And here if use boolean data type https://stackoverflow.com/questions/1740303/postgres-alter-column-integer-to-boolean/1740337
